pulsar-client-cpp/wireshark/pulsarDissector.cc

```diff
@@ -795,7 +865,7 @@ static int dissect_pulsar_message(tvbuff_t *tvb, packet_info* pinfo, proto_tree*
                 proto_tree_add_string(cmd_tree, hf_pulsar_producer_name, tvb, cmdOffset, cmdSize,
                                       success.producer_name().c_str());
 
-                proto_tree* item = proto_tree_add_uint64(cmd_tree, hf_pulsar_producer_id, tvb,
+                item = proto_tree_add_uint64(cmd_tree, hf_pulsar_producer_id, tvb,
```

Comment:
C++ allows sharing the same variable name inside and outside the `if` / `for` block. Like

```c++
    int x = 0;
    if (x == 0) {
        int x = 100;
        std::cout << x << std::endl;
    }
```

The same variable in the code block will shadow the variable outside the code block.

The original code is better because it avoids accessing the global variable, whose definition is somewhere far and might be modified.

Reply:
Right, the original code has a variable defined here, it never uses it.
```c++
    proto_tree* item = NULL;

    switch (command.type()) {
```

Reply:
Yeah, we can open another PR to fix the code style.